### DISCUSSION TITLE
Personal Access Token and privacy implications

### DISCUSSION THREAD

**Discussion by fbdb** [ImranR98/Obtainium#1329] [Q&A] [answered]
Hi,

I added a few of my apps and got very quickly rate limited by Github. The solution you provide is to add a Personal Access Token, which I could do since I have a Github account.

But before doing it, I want to be sure to understand the privacy implications : with such a token, Github could in theory know every app that I (i.e. my github account) have on my phone, right ?

I'm not sure how PAT works, hence my question.

Thank you !

**Comment by ImranR98** [comment, marked as answer]
Yes, in theory every API request you make with your token can be associated with any identifying info in your GitHub account (like your email). That said, the GitHub API has lots of use cases so it's not like they would know that "these are the mobile apps @fbdb updates through Obtainium" - it would be more like "@fbdb requests info about these repos X times a day for some reason." I also don't think GitHub accounts need phone numbers or anything like that, so you could just use a separate account with a pseudonymous email (not that that would guarantee anything).

**Reply by fbdb** [reply]
Understood, thank you for the explanation !

**Reply by e-t-l** [reply]
For Github API tokens, what is the most limited token users can create to work with Obtainium? Would it be a Fine-Grained Access token, or a Classic token? Do any access permissions need to be granted?

**Reply by warreth** [reply]
@e-t-l For anyone reading this in the future: yes you can use Fine-grained personal access tokens

**Reply by NoseyNick** [reply]
Fine-grained, "Public repositories = Read-only access to public repositories", no special repo access, no special user/repo permissions. **Set some sensible expiry date** - default 30 days may annoy you in a month!   :laughing: 
Seems to work.  :man_shrugging:

**Comment by NoseyNick** [comment]
Would be nice to add to https://wiki.obtainium.imranr.dev/sources/#creating-a-github-personal-access-token but I can't work out how to - does not seem to be a wiki in the usual sense of the word / may require a login somewhere?
